applications/FluidDynamicsApplication/custom_strategies/strategies/compressible_navier_stokes_explicit_solving_strategy_runge_kutta.h

```diff
@@ -211,16 +211,67 @@ class CompressibleNavierStokesExplicitSolvingStrategyRungeKutta
     {
         // Base class assign settings call
         BaseType::AssignSettings(ThisParameters);
-
-        // Set the specific compressible NS settings
-        mShockCapturing = ThisParameters["shock_capturing"].GetBool();
         mCalculateNonConservativeMagnitudes = ThisParameters["calculate_non_conservative_magnitudes"].GetBool();
+
+        SetUpShockCapturing(ThisParameters["shock_capturing_settings"]);
+
         if (mShockCapturing && !mCalculateNonConservativeMagnitudes) {
-            KRATOS_WARNING("CompressibleNavierStokesExplicitSolvingStrategyRungeKutta") << "\'shock_capturing\' requires \'calculate_non_conservative_magnitudes\' to be active. Activating it." << std::endl;
+            KRATOS_WARNING("CompressibleNavierStokesExplicitSolvingStrategyRungeKutta4")
+                << "\'shock_capturing\' requires \'calculate_non_conservative_magnitudes\' to be active. Activating it." << std::endl;
             mCalculateNonConservativeMagnitudes = true;
         }
     }
 
+    void SetUpShockCapturing(Parameters ShockCapturingParameters)
+    {
+        KRATOS_TRY
+
+        auto defaults = Parameters(R"(
+            {
+                "type" : "physics_based",
+                "Parameters" : {
+                    "model_part_name" : ""
+                }
+            }
+        )");
+        defaults["Parameters"]["model_part_name"].SetString(BaseType::GetModelPart().Name());
+
+        ShockCapturingParameters.ValidateAndAssignDefaults(defaults);
+        ShockCapturingParameters.RecursivelyAddMissingParameters(defaults);
+
+        const static std::map<const std::string, ShockCapturingFactoryType> shock_capturing_factory_map
+        {
+            {"none"         , [](ModelPart& m, Parameters p) -> Process::UniquePointer {return nullptr;}},
+            {"physics_based", [](ModelPart& m, Parameters p) -> Process::UniquePointer {return Kratos::make_unique<ShockCapturingPhysicsBasedProcess>(m, p);}},
+            {"entropy_based", [](ModelPart& m, Parameters p) -> Process::UniquePointer {return Kratos::make_unique<ShockCapturingEntropyViscosityProcess>(m, p);}}
+        };
+
+        const auto sc_type = ShockCapturingParameters["type"].GetString();
+
```

Comment:
You can avoid the try-catch by doing
```c++
if (shock_capturing_factory_map[sc_type] != shock_capturing_factory_map.end()) {
    ... do stuff...
} else {
    KRATOS_ERROR << "bla bla" << std::endl;
}
```

Reply:
Well the [] operator requires the maps contents to be default constructible, but using find I can also avoid the try-catch.

Reply:
Done